From 75d93e8ba34ea7b47e7890223164a89a4cbea1c1 Mon Sep 17 00:00:00 2001
From: "Ricardo B. Marlière" <rbm@opensuse.org>
Date: Thu, 6 Feb 2025 16:28:04 -0300
Subject: [PATCH] Fix shebangs to remove `env` dependency
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Get rid of the following warning:

python311-lieer.noarch: E: env-script-interpreter (Badness: 9)
/usr/lib/python3.11/site-packages/lieer/gmailieer.py /usr/bin/env python3
python311-lieer.noarch: E: env-script-interpreter (Badness: 9)
/usr/lib/python3.11/site-packages/lieer/nobar.py /usr/bin/env python3

This script uses 'env' as an interpreter. For the rpm runtime dependency
detection to work, the shebang #!/usr/bin/env <interpreter>  needs to be
patched into #!/usr/bin/<interpreter>  otherwise the package dependency
generator merely adds a dependency on /usr/bin/env rather than the actual
interpreter /usr/bin/<interpreter>.  Alternatively, if the file should not
be executed, then ensure that it is not marked as executable or don't
install it in a path that is reserved for executables.

Signed-off-by: Ricardo B. Marlière <rbm@opensuse.org>
---
 gmi                | 2 +-
 lieer/gmailieer.py | 2 +-
 lieer/nobar.py     | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/gmi b/gmi
index c3e65691cf84..b2df42b66eb3 100755
--- a/gmi
+++ b/gmi
@@ -1,4 +1,4 @@
-#! /usr/bin/env python3
+#! /usr/bin/python3
 #
 # Copyright © 2020  Gaute Hope <eg@gaute.vetsj.com>
 #
diff --git a/lieer/gmailieer.py b/lieer/gmailieer.py
index 381ab4f2be50..a719f91bdf75 100755
--- a/lieer/gmailieer.py
+++ b/lieer/gmailieer.py
@@ -1,4 +1,4 @@
-#! /usr/bin/env python3
+#! /usr/bin/python3
 #
 # Copyright © 2020  Gaute Hope <eg@gaute.vetsj.com>
 # Author: Gaute Hope <eg@gaute.vetsj.com> / 2017-03-05
diff --git a/lieer/nobar.py b/lieer/nobar.py
index 347816de008b..03d1fd35be05 100644
--- a/lieer/nobar.py
+++ b/lieer/nobar.py
@@ -1,4 +1,4 @@
-#! /usr/bin/env python3
+#! /usr/bin/python3
 #
 # Regular non-TTY drop-in replacement for tqdm
 #
-- 
2.48.1

